src/index.js

```diff
@@ -1,390 +1,604 @@
-        removeChildren(element.childNodes[i], node.children[i])
-      }
+  patchElement(container, element, lastNode, nextNode, lifecycle, eventProxy)
 
-      if (attributes.ondestroy) {
-        attributes.ondestroy(element)
-      }
-    }
-    return element
-  }
+  while (lifecycle.length > 0) lifecycle.pop()()
 
-  function removeElement(parent, element, node) {
-    function done() {
-      parent.removeChild(removeChildren(element, node))
-    }
+  return nextNode
+}
 
-    var cb = node.attributes && node.attributes.onremove
-    if (cb) {
-      cb(element, done)
-    } else {
-      done()
+export var h = function(name, props) {
+  var node
+  var rest = []
+  var children = []
+  var length = arguments.length
+
+  while (length-- > 2) rest.push(arguments[length])
+
+  if ((props = props == null ? {} : props).children != null) {
+    if (rest.length <= 0) {
+      rest.push(props.children)
     }
+    delete props.children
   }
 
-  function patch(parent, element, oldNode, node, isSvg) {
-    if (node === oldNode) {
-    } else if (oldNode == null || oldNode.nodeName !== node.nodeName) {
-      var newElement = createElement(node, isSvg)
-      parent.insertBefore(newElement, element)
-
-      if (oldNode != null) {
-        removeElement(parent, element, oldNode)
+  while (rest.length > 0) {
+    if (isArray((node = rest.pop()))) {
+      for (length = node.length; length-- > 0; ) {
+        rest.push(node[length])
       }
-
-      element = newElement
-    } else if (oldNode.nodeName == null) {
-      element.nodeValue = node
+    } else if (node === false || node === true || node == null) {
     } else {
-      updateElement(
-        element,
-        oldNode.attributes,
-        node.attributes,
-        (isSvg = isSvg || node.nodeName === "svg")
-      )
+      children.push(typeof node === "object" ? node : createTextVNode(node))
+    }
+  }
 
-      var oldKeyed = {}
-      var newKeyed = {}
-      var oldElements = []
-      var oldChildren = oldNode.children
-      var children = node.children
+  return typeof name === "function"
+    ? name(props, (props.children = children))
+    : createVNode(name, props, children, null, props.key, DEFAULT)
+}
 
-      for (var i = 0; i < oldChildren.length; i++) {
-        oldElements[i] = element.childNodes[i]
+var cancel = function(sub) {
+  sub.cancel()
+}
 
-        var oldKey = getKey(oldChildren[i])
-        if (oldKey != null) {
-          oldKeyed[oldKey] = [oldElements[i], oldChildren[i]]
-        }
+var update = function(sub, oldSub, dispatch) {
+  if (sub.effect === oldSub.effect) {
+    for (var k in merge(sub.props, oldSub.props)) {
+      if (sub.props[k] !== oldSub.props[k]) {
+        cancel(oldSub)
+        return start(sub, dispatch)
       }
+    }
+  }
+  return oldSub
+}
 
-      var i = 0
-      var k = 0
+var start = function(sub, dispatch) {
+  return merge(sub, {
+    cancel: sub.effect(sub.props, dispatch)
```

Comment:
Why are subs still using `.props` but FX aren't? This comment appears wrong: https://github.com/hyperapp/hyperapp/issues/748#issuecomment-415935138.

Reply:
Yup, still working on it! 🙇